.github/workflows/trivy_scans.yaml

```diff
@@ -0,0 +1,79 @@
+name: IaC and container image scans
+permissions: {}
+
+# Reports are uploaded to GitHub Advanced Security if the check was run for the
+# `main` or `radix` branch. In all other cases, the jobs fail if errors are
+# found and the errors are reported in a human-readable format. This is done to
+# avoid cluttering the security panel on GitHub.
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main, radix]
+  workflow_dispatch:
+
+  schedule:
+  # Run at 02:36 AM (UTC) every Monday, Wednesday, and Friday morning to ensure
+  # that containers are scanned regularly and reports are available on a work
+  # day.
+  - cron: '36 2 * * 1,3,5'
+
+env:
+  UPLOAD_REPORT: ${{ github.ref == 'refs/heads/main' || github.ref == 'refs/heads/radix' || github.event_name == 'schedule' }}
+
+jobs:
+  iac_file_scan:
+    name: Scan inftrastructure-as-code config files
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    permissions:
+      security-events: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+        with:
+          persist-credentials: false
+          ref: ${{ github.event_name ==  'schedule' && 'radix' || '' }}
+
+      - name: Scan configuration files
+        uses: "./.github/actions/trivy_scans"
+        with:
+          upload_to_ghas: ${{ env.UPLOAD_REPORT == 'true' }}
+          scan_type: 'config'
+
+  container_image_scan:
+    name: Scan Docker image
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
```

Comment:
Took me 9 minutes to run, so not sure if we shouldn't increase time-out at least to 15 minutes.